src/neo4j_genai/components/entity_relation_extractor.py

```diff
@@ -0,0 +1,114 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import enum
+import json
+import logging
+from typing import Any, Dict, Union
+
+from pydantic import BaseModel, ValidationError, validate_call
+
+from neo4j_genai.components.types import Neo4jGraph, TextChunks
+from neo4j_genai.exceptions import LLMGenerationError
+from neo4j_genai.generation.prompts import ERExtractionTemplate, PromptTemplate
+from neo4j_genai.llm import LLMInterface
+from neo4j_genai.pipeline.component import Component
+
+logger = logging.getLogger(__name__)
+
+
+class EntityRelationExtractor(Component):
+    async def run(self, chunks: TextChunks, **kwargs: Any) -> Neo4jGraph:
```

Comment:
Should this not be an abstract method?

Reply:
Absolutely: https://github.com/neo4j/neo4j-genai-python/pull/85/commits/9483d7adf7e54ccf857469ef98a8538acc67761c